src/wr_logic_longrange/nodes/finder.py

```diff
@@ -336,6 +218,80 @@ def get_valley(
+    @param hist (list): The LiDAR reading data as a list
+    @return List[List[int]]: The list of left bounds and right bounds on each obstacle
+    """
+    # this will be an array of two values with the beginning and ending index of the obstacle.
+    obstacle_list = list()
+
+    one_obstacle = []
+    for i in range(len(hist)):
+        # This prevents single noisy points from blocking out large portions of the drive window
+        if hist[i] < threshold:
+            one_obstacle.append(i)
+        elif len(one_obstacle) > 1:
+            left_bound = len(hist)
+            right_bound = 0
+            for i in range(len(one_obstacle)):
+                # Calculate size of anti-window and add to obstacle bounds
+                # pass in distance to target to caculate angle that allows robot to pass through
+                angleToIncrease = calculate_anti_window(
+                    hist[one_obstacle[i]])/sector_angle
+
+                # Update left and right bound
+                left_bound = max(min(left_bound, one_obstacle[i] - angleToIncrease), 0)
+                right_bound = min(
+                    max(right_bound, one_obstacle[i] + angleToIncrease), len(hist)
+                )
+
+            # Check to see if the obstacle we just found can actually be merged with a previous obstacle
+            while len(obstacle_list) > 0 and obstacle_list[-1][1] >= left_bound:
+                left_bound = min(left_bound, obstacle_list[-1][0])
+                right_bound = max(right_bound, obstacle_list[-1][1])
+                del obstacle_list[-1]
+            obstacle_list.append([left_bound, right_bound])
+            one_obstacle.clear()
+
+    # TODO (@bennowotny ): This code is the same as what's in the loop, so it should be abstracted out to its own function
+    if len(one_obstacle) != 0:
+        left_bound = len(hist)
+        right_bound = 0
+        for i in range(len(one_obstacle)):
+            # Calculate size of anti-window and add to obstacle bounds
+            # pass in distance to target to caculate angle that allows robot to pass through
+            angleToIncrease = calculate_anti_window(
+                hist[one_obstacle[i]])/sector_angle
+
+            # Update left and right bound
+            left_bound = max(min(left_bound, one_obstacle[i] - angleToIncrease), 0)
+            right_bound = min(
+                max(right_bound, one_obstacle[i] + angleToIncrease), len(hist)
+            )
+
+        # Check to see if the obstacle we just found can actually be merged with a previous obstacle
+        while len(obstacle_list) > 0 and obstacle_list[-1][1] >= left_bound:
+            left_bound = min(left_bound, obstacle_list[-1][0])
+            right_bound = max(right_bound, obstacle_list[-1][1])
+            del obstacle_list[-1]
+        obstacle_list.append([left_bound, right_bound])
+        one_obstacle.clear()
```

Comment:
The highlighted code here is to make sure that if an obstacle is not "closed" at the end of the input histogram list, we will still append it to the obstacle list. However, the code is identical to what is inside the above for loop, and ideally we should collapse it into the while loop above instead of having repeated code.